qupulse/pulses/constant_pulse_template.py

```diff
@@ -0,0 +1,134 @@
+"""This module defines the ConstantPulseTemplate, a pulse tempalte representating a pulse with constant values on all channels
+
+Classes:
+    - ConstantPulseTemplate: Defines a pulse via channel-value pairs
+    - ConstantPulseWaveform: A waveform instantiated from a TablePulseTemplate by providing values for its
+        declared parameters.
+"""
+
+from typing import Union, Dict, List, Set, Optional, Any, Tuple, Sequence, NamedTuple, Callable
+import numbers
+import logging
+import numpy as np
+
+from qupulse.pulses.pulse_template import ChannelID, Transformation, AtomicPulseTemplate, Loop, PulseTemplate
+from qupulse.utils.types import ChannelID
+from qupulse.pulses.parameters import ParameterNotProvidedException, ParameterConstraint, ParameterConstrainer
+from qupulse.pulses.pulse_template import AtomicPulseTemplate, MeasurementDeclaration, PulseTemplate
+from qupulse._program.waveforms import ConstantWaveform
+from qupulse.expressions import ExpressionScalar
+from qupulse.pulses.multi_channel_pulse_template import MultiChannelWaveform
+from qupulse.parameter_scope import Scope
+
+
+__all__ = ["ConstantPulseTemplate"]
+
+
+class ConstantPulseTemplate(AtomicPulseTemplate):  # type: ignore
+
+    def __init__(self, duration: float, amplitude_dict: Dict[str, Any], identifier: Optional[str] = None,
+                 name: Optional[str] = None, measurements: Optional[List[MeasurementDeclaration]] = [], **kwargs: Any) -> None:
```

Comment:
Is there a reason to use name instead of identifier?

Reply:
I believe the identifier should be unique the moment the template is serialized. Since several of my pulses have the same name, I added the `name`

Reply:
Ah, so its more a human readable tag? What do you use it for? Do you think it makes sense to extend this to other pulse templates?